sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -277,6 +282,74 @@ private Mono<ConfidentialClientApplication> getConfidentialClientApplication() {
         });
     }
 
+    private Mono<ConfidentialClientApplication> getManagedIdentityConfidentialClient() {
+        return Mono.defer(() -> {
+            String authorityUrl = TRAILING_FORWARD_SLASHES.matcher(options.getAuthorityHost()).replaceAll("")
+                + "/" + tenantId;
+
+            // Temporarily pass in Dummy Client secret and Client ID. until MSal removes its requirements.
```

Comment:
do we have an issue tracking this we can link?

Reply:
it is a topic for discussion at their next meeting, 
Depending on outcome an issue will be created by msal team.